compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -755,26 +755,29 @@ class Typer extends Namer
 
   def typedFunctionType(tree: untpd.Function, pt: Type)(implicit ctx: Context): Tree = {
```

Comment:
Sadly this constructs a tree out of our beautiful `FunctionWithMods`. The tree is something like `AppliedType(TypeConstructor=LocalImplicitFunction1,Int,Int)`).